scripts/test-e2e-local-clean.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+'use strict';
+
+/*
+ * This script, paired with test-e2e-local.js, is the full suite of
+ * tooling needed for a successful local testing experience.
+ * This script is an helper to clean up the environment fully
+ * before running the test suite.
+ *
+ * You should use this when switching between branches.
+ *
+ * It will:
+ *  - clean up node modules
+ *  - clean up the build folder (derived data, gradlew cleanAll)
+ *  - clean up the pods folder for RNTester (pod install) (and Podfile.lock too)
+ *  - kill all packagers
+ *  - remove RNTestProject folder
+ *
+ * an improvements to consider:
+ *   - an option to uninstall the apps (RNTester, RNTestProject) from emulators
+ */
+
+const {exec, exit} = require('shelljs');
+
+const {isPackagerRunning} = require('./testing-utils');
+
+console.info('\n** Starting the clean up process **\n');
+
+// let's check if Metro is already running, if it is let's kill it and start fresh
+if (isPackagerRunning() === 'running') {
+  exec("lsof -i :8081 | grep LISTEN | /usr/bin/awk '{print $2}' | xargs kill");
+  console.info('\n** Killed Metro **\n');
+}
+
+// Android
+console.info('\n** Cleaning Gradle build artifacts **\n');
+exec('./gradlew cleanAll');
+
+// iOS
+console.info('\n** Nuking the derived data folder **\n');
+exec('rm -rf ~/Library/Developer/Xcode/DerivedData');
+
+// RNTester Pods
+console.info('\n** Removing the RNTester Pods **\n');
+exec('rm -rf packages/rn-tester/Pods');
+
+// I'm not sure we want to also remove the lock file
+// exec('rm -rf packages/rn-tester/Podfile.lock');
+
+// RNTestProject
+console.info('\n** Removing the RNTestProject folder **\n');
+exec('rm -rf /tmp/RNTestProject');
+
+// final clean up
+console.info('\n** Final git level wipe **\n');
+// clean unstaged changes from git
+exec('git checkout -- .');
+// remove all the untracked files
+exec('git clean -fdx');
```

Comment:
The Android clean can be omitted if we run this.

Reply:
I see - but I kind of want to have it "spelled out" as a reference (= if someone checks out this script and sees that I use gradle CleanAll, they will know/think/consider it the better command for doing android-side cleaning)